lib/permutations.rb

```diff
@@ -1,4 +1,27 @@
 
 def permutations?(string1, string2)
-  raise NotImplementedError, "permutations? not implemented"
+
+  # if strings are not same length, we can assume they aern't permutations of one another
+  if string1.length > string2.length || string2.length > string1.length
+    return false
+  end 
+
+  # convert strings to arrays
+  string1_array = string1.split("")
+  string2_array = string2.split("")
+
+  # store letters of string1 in a hash
+  hash = {}
+  string1_array.each do |char|
+    hash[char] = true
+  end
```

Comment:
You also need to keep track of the number of each letter, not just which letters appear, such as `heelo` and `hello`.